src/shared/codelens/localLambdaRunner.ts

```diff
@@ -493,12 +483,27 @@ export const invokeLambdaFunction = async (params: {
         if (params.onWillAttachDebugger) {
             await params.onWillAttachDebugger()
         }
+
         await attachDebugger({
-            channeLogger: params.channelLogger,
             debugConfig: params.debugConfig,
-            telemetryService: params.telemetryService,
-            startInvokeMillis: startInvokeTime.getTime(),
-            runtime: params.runtime,
+            maxAttempts: MAX_DEBUGGER_ATTEMPTS,
+            channelLogger: params.channelLogger,
+            onRecordAttachDebuggerMetric: (
+                attachResult: boolean | undefined, attempts: number, attachResultDate: Date
+            ): void => {
+                recordAttachDebuggerMetric({
+                    telemetryService: params.telemetryService,
+                    result: attachResult,
+                    attempts,
+                    durationMillis: attachResultDate.getTime() - startInvokeTime.getTime(),
+                    runtime: params.runtime,
+                })
+            },
+            onWillRetry: async (): Promise<void> => {
```

Comment:
I'm not the biggest fan of `onWillRetry ` used in this way. 
1. It appears to be the same in each case. Consider making this the default implementation so it doesn't need to be specified unless non-standard behavior is desired.
2. It is not obvious so other developers that  onWillRetry is intended to specify the delay and that if no delay is provided, it would retry without pausing.

Consider eliminating `onWillRetry` here since it does not appear to be needed. i.e. a variable to which the same values is always assigned is essentially a constant.